### ISSUE TITLE
Method name clash when dealing with path parameters

### ISSUE THREAD

**Issue by DavideRossi** [mulesoft-labs/raml-for-jax-rs#314]
Two resources with the following paths:
```
/foo:
    /bar:
        get:
    /bar/{id}:
        get:
```
generate two methods with the same signature:

```
@GET
@Path("/foo/bar/{id}")   // or @Path("/foo/bar") for the first resource 
@Produces("application/json")
GetBazFooBarResponse getBazFooBar();
```

To me that is related to #313 since the application of the path parameters in the methods would lead to different signatures (and no clashing).

I'm using the maven plugin (v 2.1.0).

**Comment by jpbelang** [comment]
I'll check why this breaks without parameters.

**Comment by amagnus** [comment]
@DavideRossi I had a similar issue, and fixed by adding `uriParameters`, like the following:

```
/foo:
    /bar:
        get:
    /bar/{id}:
      uriParameters:
        id:
          type: string
      get:
```

**Comment by jpbelang** [comment]
Fixed in release/2.2.0
